helperFunctions/subscribeHelpers/InitiateSalesforceSessionHelpers/CheckChatStatusHelper.ts

```diff
@@ -83,7 +86,12 @@ export class CheckChatStatus {
 						const chasitorIdleTimeout = chatEstablishedMessage.chasitorIdleTimeout || false;
 						const sneakPeekEnabled = chatEstablishedMessage.sneakPeekEnabled;
 						const { id, persisantAffinity, persistantKey } = await retrievePersistentTokens(this.read, this.assoc);
-						const salesforceAgentName = chatEstablishedMessage.name;
+						let salesforceAgentName;
+						for (const msg of messageArray) {
+							if (msg.message.name) {
+								salesforceAgentName = msg.message.name;
+							}
+						}
```

Comment:
We are also accessing `chasitorIdleTimeout` and `sneakPeekEnabled` from the `chatEstablishedMessage`. Shouldn't we also add those?